### PR TITLE
Improves sharing and playing plio after publishing

### LINKED ISSUE #371: Make sharing and playing plio better

### LINKED ISSUE #373: Editor keeps showing the option to publish changes upon reloading published plio page without having made any changes

### LINKED ISSUE #376: More tests for frontend

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #371 
Fixes #373 
Fixes #376 

## Summary
- many new tests
- share button in each plioListItem and in editor
- share dialog that opens on clicking share button
- social sharing on twitter and whatsapp + can copy link
- dialog after publishing with option to share plio / play / go to home
- dialogs are removing upon clicking away from dialog
- confetti celebration on publishing plio
- added accordion to plioListItem if screen size < 420 to toggle visibility of action buttons in plioListItem
- the action items in the first plio in the table are shown by default so that users know that they can toggle visibility of action buttons.
- editor used to make the publish button active simply on reloading - resolved that.

## Test Plan
- [x] Test Responsiveness
   - [ ] Laptop (1200px)
   - [ ] Tablet (760px)
   - [ ] Phone (320px)
- [x] Cross-Browser Testing
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Safari
- [x] Local Language Support
- [x] Wrote tests
- [x] Tested locally
- [x] Tested on staging
- [x] Tested on an actual physical phone
- [ ] Tested on production
- [x] ~Check for bundle size [here](https://bundlephobia.com/) if adding a package~
- [x] Added relevant details like Labels/Projects/Milestones etc.

### CODE REVIEW

File: tests/unit/components/Collections/Table/Table.spec.js [hunk 1]
```diff
@@ -1,9 +1,240 @@
...
+  },
+];
+
+const tableColumns = ["name", "number_of_viewers"];
+const totalNumberOfPlios = dummyTableData.length;
 
 describe("Table.vue", () => {
+  beforeEach(async () => {
+    await store.dispatch("sync/stopLoading");
+    await setMatchMedia(false);
```
Comment (deepansh96) [inline tests/unit/components/Collections/Table/Table.spec.js:23]: what does this do?

File: tests/unit/components/Collections/Table/Table.spec.js [hunk 2]
```diff
@@ -1,9 +1,240 @@
...
+      },
+    });
+    // the button is rendered for each columns but is hidden on all columns
+    // except the last column
+    expect(
+      wrapper.findAll('[data-test="analyzeButton"]')[1].element.disabled
+    ).toBe(true);
+    expect(
+      wrapper.findAll('[data-test="analyzeButton"]')[3].element.disabled
+    ).toBe(false);
```
Comment (deepansh96) [inline tests/unit/components/Collections/Table/Table.spec.js:96]: please explain

File: tests/unit/components/Items/Question/Footer.spec.js [hunk 3]
```diff
@@ -2,7 +2,11 @@ import { mount } from "@vue/test-utils";
 import Footer from "@/components/Items/Question/Footer";
 
 describe("Footer.vue", () => {
-  let wrapper = mount(Footer);
+  let wrapper;
+  beforeEach(() => {
+    wrapper = mount(Footer);
+  });
```
Comment (deepansh96) [inline tests/unit/components/Items/Question/Footer.spec.js:8]: why beforeEach here?

File: tests/unit/components/Items/Question/Header.spec.js [hunk 4]
```diff
@@ -6,4 +6,69 @@ describe("Header.vue", () => {
...
   });
+
+  it("should render when modal minimized", () => {
+    const wrapper = mount(Header, {
+      props: {
+        isModalMinimized: true,
+      },
+    });
+    expect(wrapper).toBeTruthy();
+  });
```
Comment (deepansh96) [inline tests/unit/components/Items/Question/Header.spec.js:17]: but the header component is not there when the modal is minimised right?

File: src/App.vue [hunk 5]
```diff
@@ -151,6 +161,10 @@ export default {
     UserConfigService.setLocaleFromUserConfig();
   },
   watch: {
+    currentRoute() {
+      // unset the share plio variable if it was set as true when the app is being loaded
```
[the remark this answers is not in the document]
Comment (deepansh96) [reply src/App.vue:165]: Also, here there's no need to check for `if (this.isSharePlioDialogShown)` right?
We should unset it regardless whenever the currentRoute changes

File: src/components/App/SharePlioDialog.vue [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    shareOnWhatsApp() {
+      // share the plio link on whatsapp
+      window.open("https://wa.me/send?text=" + this.socialSharingText).focus();
+    },
+    shareOnTwitter() {
+      // share the plio link on twitter
+      window
+        .open("https://twitter.com/intent/tweet?text=" + this.socialSharingText)
+        .focus();
+    },
```
[the remark this answers is not in the document]
Comment (deepansh96) [reply src/components/App/SharePlioDialog.vue:149]: Not super necessary though
When we have more sharing options, then to keep the code concise we might have to combine them into 1 method. If it's not too much work we can do that on this PR itself

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: src/components/Collections/ListItems/PlioListItem.vue [hunk 7]
```diff
@@ -250,6 +289,26 @@ export default {
   methods: {
     ...mapActions("sync", ["startLoading", "stopLoading"]),
     ...mapActions("plioItems", ["fetchPlio"]),
+    ...mapActions("generic", ["showSharePlioDialog"]),
+    ...Utilities,
+    toggleActionButtonVisibility() {
+      // toggles the visibility of the action buttons
+      this.showActionButtons = !this.showActionButtons;
+      this.isActionButtonVisibilitySet = true;
```
[the remark this answers is not in the document]
Comment (dalmia) [reply src/components/Collections/ListItems/PlioListItem.vue:297]: hasUserSetActionVisibility

File: tests/unit/components/Collections/Table/Table.spec.js (hunk 2 above)
Comment (dalmia) [reply tests/unit/components/Collections/Table/Table.spec.js:96]: comment saying draft - disabled, published - not disabled

Comment (dalmia) [reply tests/unit/components/Collections/Table/Table.spec.js:96]: num rows (2) x num cols (2)

**Pull request merged**